Better Mailchimp logic

IvanGrginovInf opened the pull request:
Mailchimp route only adds users by default (doesn't update them).

We should never really have a form which updates on frontend (unless it's specifically a form for updating user data - which is currently out of scope).

If user already exists it still returns 200 with an appropriate message "User already exists" (as that's not really an error).

dingo-d reviewed (commented):
The error that PHPStan warns you about is something you should address. The getResponse can be null: https://github.com/guzzle/guzzle/blob/34bc199aef77ff1c13f7126c0464f8fa9ee55f9f/src/Exception/RequestException.php#L141

IvanGrginovInf commented:
> The error that PHPStan warns you about is something you should address
@dingo-d Yup, didn't have time to run the tests / analyze when I opened this but had to open a PR so I don't forget to merge the changes (since I was changing this in a project).

Fixed phpstan errors, fixed tests, in the process found and fixed a bug, the whole 9 yards. 😄
